src/transforms/uglify.js

```diff
@@ -14,18 +15,36 @@ module.exports = async function(asset) {
     },
     compress: {
       drop_console: true
-    }
+    },
+    sourceMap: asset.options.sourceMaps
+      ? {
+          filename: asset.relativeName
+        }
```

Comment:
I think uglify has a way to pass an input source map to it so you don't have to manually map it like you did below. See https://github.com/mishoo/UglifyJS2#source-map-options. Looks like the `content` option?

Reply:
I've seen it but that would add an extra stringify and parse operation to the building of the new sourcemap, which is the largest performance problem with source-map 0.6.1